src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -513,12 +518,13 @@ RoverPositionControl::run()
 			//orb_copy(ORB_ID(vehicle_attitude), _vehicle_attitude_sub, &_vehicle_att);
 			_act_controls.timestamp = hrt_absolute_time();
 
-			/* Only publish if any of the proper modes are enabled */
-			if (_control_mode.flag_control_velocity_enabled ||
-			    _control_mode.flag_control_attitude_enabled ||
-			    manual_mode) {
-				/* publish the actuator controls */
-				_actuator_controls_pub.publish(_act_controls);
+			if (_actuator_controls_pub != nullptr) {
+				//PX4_INFO("Publishing actuator from pos control");
+				orb_publish(ORB_ID(actuator_controls_0), _actuator_controls_pub, &_act_controls);
+
+			} else {
+
+				_actuator_controls_pub = orb_advertise(ORB_ID(actuator_controls_0), &_act_controls);
```

Comment:
Why was this changed? Shouldn't we still only publish actuator controls when in the right mode?

Reply:
As I understand it RoverPositionControl is only called when we are actively in position mode,  see line 422

```
if (!manual_mode && _control_mode.flag_control_position_enabled) {
if (control_position(current_position, ground_speed, _pos_sp_triplet)) {
```

So didn't really understand why the flags for velocity or attitude control are checked inside the position loop, perhaps it's inherited from an older implementation and hasn't been cleaned before?

Reply:
@ealdaz-seesai This is not true, although the name of the file is rover position control, it can be used in different modes such as offboard modes. Those functions are used to support different setpoints(position, velocity, attitude, actuator) in offboard mode.